tests/backends/test_sparqlwrapper_graphdb.py

```diff
@@ -0,0 +1,138 @@
+"""
+Test the sparqlwrapper backend with GraphDB.
+Note that this requires to have a running graphDB instance set up
+as described in the for_developers documentation.
+"""
+
+import time
+
+import pytest
+import requests
+
+# URL to check if GraphDB is running.
+GRAPHDB_CHECK_URL = "http://localhost:7200/repositories"
+
+
+def graphdb_available():
+    """
+    Fixture that checks if the GraphDB instance is available.
+    If it is not, the tests that depend on it will be skipped.
+    """
+    timeout = 10  # seconds
+    interval = 1  # seconds
+    start_time = time.time()
+    while True:
+        try:
+            response = requests.get(GRAPHDB_CHECK_URL, timeout=timeout)
+            if response.status_code == 200:
+                return True
+        except requests.exceptions.RequestException:
+            pass
+
+        if time.time() - start_time > timeout:
+            return False
+        time.sleep(interval)
+
+
+def test_graphdb():
+    """
+    Fixture that creates a Triplestore instance.
+    This will only be used if the GraphDB instance is available.
+    """
+    # Check if GraphDB is available and write a warning if it is not.
+    if not graphdb_available():
+        pytest.skip("GraphDB instance not available locally; skipping tests.")
+
+    from pathlib import Path
+
+    from tripper import Literal, Triplestore
+    from tripper.datadoc import load_dict, save_datadoc, search_iris
+
+    thisdir = Path(__file__).resolve().parent
+    datasetinput = thisdir / "datadocumentation_sample.yaml"
+
+    ts = Triplestore(
+        backend="sparqlwrapper",
+        base_iri="http://localhost:7200/repositories/test_repo",
+        update_iri="http://localhost:7200/repositories/test_repo/statements",
+    )
+
+    # Test adding triples
+    ts.add_triples(
+        [
+            (
+                "http://www.example.org/subject",
+                "http://www.example.org/predicate",
+                Literal("a"),
+            ),
+            (
+                "http://www.example.org/subject",
+                "http://www.example.org/predicate",
+                Literal("b"),
```

Comment:
Maybe replace `Literal("b")` with `Literal(1.0)` to test that the backend correctly handles non-string literals.